pages/docs/reference/evolution/components-stability.md

```diff
@@ -2,41 +2,75 @@
 
-The provisions of the [Kotlin evolution policy](kotlin-evolution.html) fully apply only to components that have reached Full Stability (FS). From that point on incompatible changes have to be approved by the Language Committee.
+**Alpha** means "use at your own risk, expect migration issues": 
+  * We decided to productize this idea, but it hasn't reached the final shape yet.
 
-|**Component**|**Status Entered at version**|**Mode for Sources**|**Mode for Binaries**|
+**Beta** means "you can use it, we'll do our best to minimize migration issues for you": 
+  * It’s almost done, user feedback is especially important now.
+  * Still, it's not 100% finished, so changes are possible (including ones based on your own feedback).
+  * Watch for deprecation warnings in advance for the best update experience.
+
+We collectively refer to _Experimental_, _Alpha_ and _Beta_ as **pre-stable** levels.
+
+<a name="stable"></a>
+**Stable** means "use it even in most conservative scenarios":
+  * It’s done. We will be evolving it according to our strict [backward compatibility rules](/foundation/language-committee-guidelines.html).
+  
+Please note that stability levels do not say anything about how soon a component will be released as Stable. Similarly, they do not indicate how much a component will be changed before release. They only say how fast a component is changing and how much risk of update issues users are running.
+
+## Stability of Subcomponents
+
+A stable component may have an experimental subcomponent, for example:
+* a stable compiler may have an experimental feature;
+* a stable API may include experimental classes or functions;
+* a stable command-line tool may have experimental options.
+
+We make sure to document precisely which subcomponents are not stable. We also do our best to warn users where possible and ask to opt in explicitly to avoid accidental usages of features that have not been released as stable.
+
+## Current Stability of Kotlin Components
+
+|**Component**|**Status**|**Status since version**|**Comment**|
 | --- | --- | --- | --- |
-Kotlin/JVM|1.0|FS|FS|
-kotlin-stdlib (JVM)|1.0|FS|FS
-KDoc syntax|1.0|FS|N/A
-Coroutines|1.3|FS|FS
-kotlin-reflect (JVM)|1.0|SIR|SIR
-Kotlin/JS|1.1|AIR|MF
-Kotlin/Native|1.3|AIR|MF
-Kotlin Scripts (*.kts)|1.2|AIR|MF
-dokka|0.1|MF|N/A
-Kotlin Scripting APIs|1.2|MF|MF
-Compiler Plugin API|1.0|MF|MF
-Serialization|1.3|MF|MF
-Multiplatform Projects|1.2|MF|MF
-Inline classes|1.3|MF|MF
-Unsigned arithmetics|1.3|MF|MF
-**All other experimental features, by default**|N/A|**MF**|**MF**
\ No newline at end of file
+Kotlin/JVM|Stable|1.0|
+kotlin-stdlib (JVM)|Stable|1.0|
+Coroutines|Stable|1.3|
+kotlin-reflect (JVM)|Beta|1.0|
+Kotlin/JS (Classic back-end)|Stable|1.3|
+Kotlin/JVM (IR-based)|Alpha|1.4|
+Kotlin/JS (IR-based)|Alpha|1.4|
+kotlin-stdlib API (JS)|Stable|1.3|
```

Comment:
I can't recall the decision to make Kotlin/JS and stdlib-js stable in 1.3. It is currently listed as AIR on the site.